src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -1209,27 +1211,117 @@ void runtime::cpu::CPU_ExternalFunction::build()
+                            });
+                    nodename_tbbnode_map.insert({it->second, flowgraph_node});
+                    it++;
+                }
+
+                vector<Node*> dependence_graph_heads;
+
+                traverse_nodes(
+                    m_function,
+                    [&dependence_graph_heads, &nodename_tbbnode_map](shared_ptr<Node> n) {
+                        if (!n->is_parameter() && !n->is_constant())
+                        {
+                            bool is_head = true;
+                            for (auto arg : n->get_arguments())
+                            {
+                                if (!arg->is_parameter() && !arg->is_constant())
+                                {
+                                    is_head = false;
+                                    tbb::flow::make_edge(*(nodename_tbbnode_map[arg->get_name()]),
+                                                         *(nodename_tbbnode_map[n->get_name()]));
+                                }
+                            }
+                            if (is_head)
+                            {
+                                dependence_graph_heads.emplace_back(n.get());
+                            }
+                        }
+                    });
+
+                if (!dependence_graph_heads.empty())
+                {
+                    for (Node* n : dependence_graph_heads)
+                    {
+                        tbb::flow::make_edge(*flowgraph_node_start,
+                                             *(nodename_tbbnode_map[n->get_name()]));
+                    }
+                }
+                if (m_release_function)
                 {
-                    (*functor)(ctx);
-                    std::advance(functor, 1);
+                    release_function();
                 }
             }
-            else
+            // Execute the flow graph
+            ((tbb::flow::continue_node<tbb::flow::continue_msg, tbb::flow::lightweight>*)(&(
+                 *(ctx->G->begin()))))
+                ->try_put(tbb::flow::continue_msg());
+            try
+            {
+                ctx->G->wait_for_all();
+            }
+            catch (...)
+            {
+                throw;
+            }
+        }
+        else
+        {
+            for (const auto& p : enables)
             {
-                std::advance(functor, p.second);
+                if (p.first(ctx) || first_iteration)
+                {
+                    for (size_t j = 0; j < p.second; j++)
+                    {
+                        (*functor)(ctx);
+                        std::advance(functor, 1);
+                    }
+                }
+                else
+                {
+                    std::advance(functor, p.second);
+                }
             }
         }
         first_iteration = false;
     };
 
     m_is_built = true;
 
-    if (m_release_function)
+    if (m_release_function && !m_use_tbb)
```

Comment:
@jmenon do we need this in the executor or should this be done at the end of the build function outside the executor.

Reply:
The executor needs m_function to build the tbb flow graph at the first iteration. It is released after tbb flow graph is built.